amlip_cpp/src/cpp/dds/impl/Reader.ipp

```diff
@@ -95,9 +103,15 @@ T Reader<T>::read()
         // Set number of data still available
         if (datareader_locked_->get_unread_count() <= 0)
         {
+            // TODO: there is a race condition here as get_unread_count is not guarded by any mutex here
+            // and so a call to is_data_available could be skipped here.
```

Comment:
Right, let's discuss possible solutions.

- Use `CountWaitHandler` instead
- Use an atomic state and perform atomic transitions in a finite state machine

Reply:
Probably use a complex machine state as in DDSRouter track, as it does not require mutex or any other synchronization

Reply:
Future PR